test/Core.js

```diff
@@ -19,38 +19,205 @@
+            , parentNodes = ethUtil.rlp.decode(proof.account.rlpParentNodes)
+            , accountNode = parentNodes[parentNodes.length - 1]
+            , accountValue = ethUtil.rlp.decode(accountNode[1])
+            , storageRoot = '0x' + accountValue[2].toString('hex')
+        ;
+
+        before(async () => {
+
+            contractsData = await coreUtils.deployCore(artifacts, accounts);
+            core = contractsData.core;
+            worker = contractsData.worker;
+
+            await core.commitStateRoot(blockHeight, proof.account.stateRoot, {from: worker});
+        });
+
+        it('should not be able to verify proof for account if block height is 0', async () => {
+            await utils.expectThrow(core.proveOpenST(0, proof.account.rlpEncodedAccount, proof.account.rlpParentNodes, {from: worker}));
+        });
+
+        it('should not be able to verify proof for account if rlpEncodedAccount value is blank', async () => {
+            await utils.expectThrow(core.proveOpenST(blockHeight, '', proof.account.rlpParentNodes, {from: worker}));
+        });
+
+        it('should not be able to verify proof for account if rlpEncodedAccount value is 0x', async () => {
+            await utils.expectThrow(core.proveOpenST(blockHeight, '0x', proof.account.rlpParentNodes, {from: worker}));
+        });
+
+        it('should not be able to verify proof for account if rlpParentNodes value is blank', async () => {
+            await utils.expectThrow(core.proveOpenST(blockHeight, proof.account.rlpEncodedAccount, '', {from: worker}));
+        });
+
+        it('should not be able to verify proof for account if rlpParentNodes value is 0x', async () => {
+            await utils.expectThrow(core.proveOpenST(blockHeight, proof.account.rlpEncodedAccount, '0x', {from: worker}));
+        });
+
+        it('should be able to verify proof for account with wasAlreadyProved = false ', async () => {
+            let response = await core.proveOpenST(blockHeight, proof.account.rlpEncodedAccount, proof.account.rlpParentNodes, {from: worker});
+            let formattedDecodedEvents = web3EventsDecoder.perform(response.receipt, core.address, core.abi);
+            let event = formattedDecodedEvents['OpenSTProven'];
+            await coreUtils.checkOpenSTProvenEvent(event, blockHeight, storageRoot, false);
+        });
+
+        it('should be able to verify proof for account with wasAlreadyProved = true', async () => {
+            let response = await core.proveOpenST(blockHeight, proof.account.rlpEncodedAccount, proof.account.rlpParentNodes, {from: worker});
+            let formattedDecodedEvents = web3EventsDecoder.perform(response.receipt, core.address, core.abi);
+            let event = formattedDecodedEvents['OpenSTProven'];
+            await coreUtils.checkOpenSTProvenEvent(event, blockHeight, storageRoot, true);
+        });
+
+        it('should be able to verify proof for account if called by non worker', async () => {
```

Comment:
The answer is yes because this test specifically focuses that proveOpenST can be called by anyone. It also emphasizes that any data on value/utility chain can be proven by anyone. Also As we are moving towards whitelisted workers gradually, I think it's important to convey this point.

Will update the inputs as per your suggestion.